src/open_inwoner/openproducten/producttypes_imports.py

```diff
@@ -0,0 +1,380 @@
+import logging
+
+from django.db import transaction
+from django.utils.text import slugify
+
+from open_inwoner.openproducten.models import Price, PriceOption
+from open_inwoner.pdc.models import (
+    Category,
+    Product as ProductType,
+    ProductCondition,
+    ProductFile,
+    ProductLink,
+    Question,
+    Tag,
+    TagType,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def _update_instance(model, commit=False, **kwargs):
+    """Update a model instance by a dict"""
+    for key, value in kwargs.items():
+        setattr(model, key, value)
+    if commit:
+        model.save()
+
+
+def _get_instance(model, uuid):
+    """Returns an model instance by uuid or None"""
+    return model.objects.filter(open_producten_uuid=uuid).first()
+
+
+class OpenProductenImporter:
+    def __init__(self, client):
+        self.client = client
+        self.created_objects = []
+        self.updated_objects = []
+
+    def _add_to_log_list(self, instance, created: bool):
+        if created:
+            self.created_objects.append(instance)
+        else:
+            self.updated_objects.append(instance)
+
+    def _update_or_create_question(self, question, product_type=None, category=None):
+
+        relation_object = product_type if product_type else category
+        relation_key = "product" if product_type else "category"
+
+        question_instance, created = Question.objects.update_or_create(
+            open_producten_uuid=question.id,
+            defaults={
+                "open_producten_uuid": question.id,
+                "question": question.question,
+                "answer": question.answer,
+                relation_key: relation_object,
+            },
+        )
+        self._add_to_log_list(question_instance, created)
+
+
+class ProductTypeImporter(OpenProductenImporter):
+    def __init__(self, client):
+        super().__init__(client)
+        self.product_types = None
+        self.handled_product_types = set()
+
+    @transaction.atomic()
+    def import_producttypes(self):
+        """
+        generate a Product for every ProductType in the Open Producten API
```

Comment:
Conventie voor docstrings is volledige zinnen (caps, leestekens):

```suggestion
        Generate a Product for every ProductType in the Open Producten API.
```